src/Magento/FunctionalTestingFramework/DataGenerator/Handlers/PersistedObjectHandler.php

```diff
@@ -87,7 +87,7 @@ public function createEntity(
             $retrievedDependentObjects[] = $this->retrieveEntity($objectKey, $scope);
         }
         
-        $retrievedEntity = DataObjectHandler::getInstance()->getObject($entity);
+        $retrievedEntity = DataObjectHandler::getInstance(false)->getObject($entity);
```

Comment:
DataObjectHandler::getInstance() is used everywhere in our code. What's the rule to set the flag to true or false? Why set it to false here?

Reply:
suppressed name validation while calling `createEntity`, `getEntity`, `updateEntity`. These functions are only invoked during runtime. Looks DataObjectHandler, creates a new instance of the class during runtime which is why these validations happen again (not 100% sure, that's what it looks like). Rule probably is - suppress during runtime instantiation.

Reply:
It's not necessary to use a new flag. Let's just use `MftfApplicationConfig::EXECUTION_PHASE`

Reply:
Adding this flag in widely used framework code to avoid displaying deprecation message more than once seems not unnecessary. I agree with @okolesnyk, if we cannot avoid displaying on generation time and runtime for deprecated metadata, it's fine to display twice.

Reply:
Removed flag as per comments.